README.md

```diff
@@ -1,138 +1,128 @@
+- `color` - `THREE.Color` to paint the line width, or tint the texture with
+- `opacity` - alpha value from 0 to 1 (requires `transparent` set to `true`)
+- `alphaTest` - cutoff value from 0 to 1
+- `dashArray` - the length and space between dashes. (0 - no dash)
+- `dashOffset` - defines the location where the dash will begin. Ideal to animate the line.
+- `dashRatio` - defines the ratio between that is visible or not (0 - more visible, 1 - more invisible).
+- `resolution` - `THREE.Vector2` specifying the canvas size (REQUIRED)
+- `sizeAttenuation` - makes the line width constant regardless distance (1 unit is 1px on screen) (0 - attenuate, 1 - don't attenuate)
+- `lineWidth` - float defining width (if `sizeAttenuation` is true, it's world units; else is screen pixels)
 
-* Better miters
-* Proper sizes
+If you're rendering transparent lines or using a texture with alpha map, you should set `depthTest` to `false`, `transparent` to `true` and `blending` to an appropriate blending mode, or use `alphaTest`.
 
-### Support ###
+#### Use MeshLine and MeshLineMaterial to create a THREE.Mesh
 
-Tested successfully on
-
-* Chrome OSX, Windows, Android
-* Firefox OSX, Windows, Anroid
-* Safari OSX, iOS
-* Internet Explorer 11 (SVG and Shape demo won't work because they use Promises)
-* Opera OSX, Windows
-
-### References ###
+Finally, we create a mesh and add it to the scene:
 
-* [Drawing lines is hard](http://mattdesl.svbtle.com/drawing-lines-is-hard)
-* [WebGL rendering of solid trails](http://codeflow.org/entries/2012/aug/05/webgl-rendering-of-solid-trails/)
-* [Drawing Antialiased Lines with OpenGL](https://www.mapbox.com/blog/drawing-antialiased-lines/)
+```js
+const mesh = new THREE.Mesh(line, material)
+scene.add(mesh)
+```
 
-#### License ####
+You can optionally add raycast support with the following.
 
-MIT licensed
+```js
+mesh.raycast = MeshLineRaycast
+```
 
-Copyright (C) 2015-2016 Jaume Sanchez Elias, http://www.clicktorelease.com
+# Declarative use
+
+threejs-meshline has getters and setters that make declarative usage a little easier. This is how it would look like in react/[react-three-fiber](https://github.com/drcmda/react-three-fiber). You can try it live [here](https://codesandbox.io/s/react-three-fiber-threejs-meshline-example-vl221).
+
+```jsx
+import { extend, Canvas } from 'react-three-fiber'
+import { MeshLine, MeshLineMaterial, MeshLineRaycast } from 'threejs-meshline'
+
+extend({ MeshLine, MeshLineMaterial })
+
+function Line({ vertices, width, color }) {
+  return (
+    <Canvas>
+      <mesh raycast={MeshLineRaycast}>
+        <meshLine attach="geometry" vertices={vertices} />
+        <meshLineMaterial
+          attach="material"
+          transparent
+          depthTest={false}
+          lineWidth={width}
+          color={color}
+          dashArray={0.05}
+          dashRatio={0.95}
+        />
+      </mesh>
+    </Canvas>
+  )
+}
+```
```

Comment:
I can just make it the existing readme for the timebeing.

Reply:
I've changed the readme to match the existing one. I've adjusted the examples to match the simpler api, using `setVertices` instead of creating first geometry using vertices then using `setGeometry` to extract the vertices.

`Meshline` is now a bufferGeom so you can create a mesh using `mesh(meshline, material)` instead of `mesh(meshline.geometry, material)`

I added the declaritive `react-three-fiber` examples to the bottom of the readme.

Reply:
Amazing thanks a lot for those big changes!

I've made some minor corrections to have everything consistent but everything should be good!